types/underscore/index.d.ts

```diff
@@ -3436,36 +3438,27 @@ declare module _ {
         values(object: any): any[];
 
         /**
-         * Like map, but for objects. Transform the value of each property in turn.
-         * @param object The object to transform
-         * @param iteratee The function that transforms property values
-         * @param context The optional context (value of `this`) to bind to
-         * @return a new _.Dictionary of property values
-         */
-        mapObject<T, U>(object: _.Dictionary<T>, iteratee: (val: T, key: string, object: _.Dictionary<T>) => U, context?: any): _.Dictionary<U>;
-
-        /**
-         * Like map, but for objects. Transform the value of each property in turn.
-         * @param object The object to transform
-         * @param iteratee The function that tranforms property values
-         * @param context The optional context (value of `this`) to bind to
-         */
-        mapObject<T>(object: any, iteratee: (val: any, key: string, object: any) => T, context?: any): _.Dictionary<T>;
-
-        /**
-         * Like map, but for objects. Retrieves a property from each entry in the object, as if by _.property
-         * @param object The object to transform
-         * @param iteratee The property name to retrieve
-         * @param context The optional context (value of `this`) to bind to
+         * Like map, but for objects. Transform the value of each property in
+         * turn.
+         * @param object The object to transform.
+         * @param iteratee The iteratee to use to transform property values.
+         * @param context `this` object in `iteratee`, optional.
+         * @returns A new object with all of `object`'s property values
+         * transformed through `iteratee`.
          */
-        mapObject(object: any, iteratee: string, context?: any): _.Dictionary<any>;
+        mapObject<V extends object, I extends Iteratee<V, any, TypeOfCollection<V, any>>>(
+            object: V,
+            iteratee: I,
+            context?: any
+        ): { [K in keyof V]: IterateeResult<I, V[K]> };
 
         /**
-        * Convert an object into a list of [key, value] pairs.
-        * @param object Convert this object to a list of [key, value] pairs.
-        * @return List of [key, value] pairs on `object`.
-        **/
-        pairs(object: any): [string, any][];
+         * Converts `object` into a list of [key, value] pairs. The opposite
+         * of the single-argument signature of `_.object`.
+         * @param object The object to convert.
+         * @returns The list of [key, value] pairs from `object`.
+         **/
+        pairs<V extends object>(object: V): [string, TypeOfCollection<V, any>][];
```

Comment:
Ah, I just realized... is `[keyof V, TypeOfCollection<V, any>][]` feasible?

Reply:
100% feasible, 💯 on that call!

Reply:
I did do the same `Extract` thing as with `findKey` though.

Reply:
Just to show that `number` should still be avoided in this case:

![image](https://user-images.githubusercontent.com/8692126/89061420-88252400-d319-11ea-9ba2-dd860fbdb51b.png)